x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java

```diff
@@ -453,7 +474,8 @@ private AuthenticationResult<User> performAuthentication(
 
         final Settings.Builder settingsBuilder = Settings.builder()
             .put(getFullSettingKey(REALM_NAME, SamlRealmSettings.PRINCIPAL_ATTRIBUTE.getAttribute()), useNameId ? "nameid" : "uid")
-            .put(getFullSettingKey(REALM_NAME, SamlRealmSettings.GROUPS_ATTRIBUTE.getAttribute()), "groups")
+            .put(getFullSettingKey(REALM_NAME, SamlRealmSettings.GROUPS_ATTRIBUTE.getAttributePatternSetting().getAttribute()), "groups")
+            .put(getFullSettingKey(REALM_NAME, SamlRealmSettings.GROUPS_ATTRIBUTE.getDelimiter()), "$")
```

Comment:
I'd add an explicit parameter to toggle this behavior; otherwise we're essentially losing coverage for the case where we're _not_ using a string delimiter (`testAuthenticateWithAuthorizingRealm` doesn't cover quite the same flow I don't think).

Reply:
Yes, good idea.